Advent of code

dmitri-sirobokov opened the pull request:
Blog article, sharing experience with Advent Of Code 2022, https://adventofcode.com/2022/about

Commit:
Advent Of Code by Dmitri Sirobokov

Commit:
Update the date

DaveBitter commented on data/blog/advent-of-code.md:
```diff
@@ -0,0 +1,284 @@
+
+### Typing skills
+
+After typing hundreds of for-loops, I wish I could think and type as quickly as [Victor Rentea](https://www.youtube.com/watch?v=cK19rE2V9UY), and I wish I watched his videos before. That would save me so much time. Something to learn for the next time!
```
Perhaps embed the YouTube video? I believe it is possible

dmitri-sirobokov replied:
Done

DaveBitter commented on data/blog/advent-of-code.md:
````diff
@@ -0,0 +1,284 @@
+
+```
+
+One more thing to mention here. Starting from Day 5 I am using parameterized unit tests. I figured out how to extend my JUnit framework using ArgumentsProvider interface, I have removed duplicate code that reads sample files for my test cases. My code became at least 2x shorter and cleaner after that:
````
Perhaps link to some of the resources you used to learn about it?

dmitri-sirobokov replied:
Done

DaveBitter commented on data/blog/advent-of-code.md:
```diff
@@ -0,0 +1,284 @@
+
+## Classes of problems
+
+## Integer overflow
```
It looks like you heading levels are off. There are two h2's after each other here

dmitri-sirobokov replied:
Fixed.

DaveBitter reviewed (commented):
Great article @dmitri-sirobokov!

Commit:
- Add a paragraph explaining puzzle structure
- embedded video of Victor Rentea
- Link to the guide parameterized unit tests
- fix heading levels

DaveBitter commented:
LGTM, @sanderdejong88 ?
